src/routes/upload/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { ReactElement } from 'react'
+import { Grid, Input, Button, Select } from '@material-ui/core'
+import NavBar from '../../components/NavBar'
+import { ENDPOINT } from '../../utils/config'
+import FileInput from '../../components/FileInput'
+import './main.sass'
+
+function isLIURL(str: string) {
+  const pattern = new RegExp(
+    '^(https?:\\/\\/)?' + // protocol
+    '(((linkedin)|(www)).)+com/in' + // domain name
+      '(\\/[-a-z\\d%_.~+]+)$', // port and path
+    'i'
+  ) // fragment locator
+  return !!pattern.test(str)
+}
+
+function isValidLIUser(str: string) {
+  const pattern = new RegExp('^[a-zA-Z\\d-]{3,100}$', 'i') // fragment locator
+  return !!pattern.test(str)
+}
+
+const LabelInput = (props: {
+  name: string
+  label: string
+  type?: string
+  children?: ReactElement
+  className?: string
+}) => {
+  return (
+    <label className="dg-form-input MuiFormControlLabel-label MuiTypography-body1 MuiFormControlLabel-root">
+      {`${props.label}: `}
+      {props.children ?? <Input required type={props.type ?? 'text'} name={props.name} />}
+    </label>
+  )
+}
+
+class App extends React.Component<{ classes: any }> {
+  state = {
+    error: '',
+    success: false,
+    majorOther: false,
+  }
+
+  majorWrapper: React.Ref<any>
+
+  constructor(props: any) {
+    super(props)
+    this.majorWrapper = React.createRef()
+
+    this.checkOther = this.checkOther.bind(this)
+  }
+
+  toBase64(file: File): Promise<null | string | ArrayBuffer> {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader()
+      reader.readAsDataURL(file)
+      reader.onload = () => resolve(reader.result)
+      reader.onerror = (error) => reject(error)
+    })
+  }
+
+  normalizeData(event: React.ChangeEvent<any>) {
+    const name = event.target.name.value.trim()
+    const email = event.target.email.value.trim()
+    const gpa = event.target.gpa.value.trim()
+    const standing = event.target.standing.value.trim()
+
+    let linkedIn = event.target.linkedin.value.trim().toLowerCase()
+    if (isLIURL(linkedIn)) {
+      const indexWWW = linkedIn.indexOf('www.')
+      const indexHTTPS = linkedIn.indexOf('https://')
+      const indexLinkedIn = linkedIn.indexOf('linkedin')
+      const httpsLength = 'https://'.length
+
+      if (indexWWW === -1 || indexWWW > indexLinkedIn) {
+        if (indexHTTPS !== -1) {
+          linkedIn =
+            linkedIn.substring(0, httpsLength) + 'www.' + linkedIn.substring(httpsLength)
+        } else {
+          linkedIn = 'www.' + linkedIn
+        }
+      }
+
+      if (indexHTTPS === -1) {
+        linkedIn = 'https://' + linkedIn
+      }
```

Comment:
```suggestion
    if (isLIURL(linkedIn)) {
      const indexWWW = linkedIn.indexOf('www.')
      const indexHTTPS = linkedIn.indexOf('https://')

      if (indexWWW === -1 )
         linkedin = 'www.' + linkedin
      if (indexHTTPS !== -1) 
          linkedIn = 'https://' + linkedIn
```

Reply:
This does not account for the case of https://linkedin.com/in/ddd
It would produce the following: www.https://linkedin.com/in/ddd
This did help me realize that I did forget the http:// case

Reply:
oh I also thought this does not account for linkedin.com/in/.www case, but I realized that that is checked with isLIURL